perflib/perflib.go

```diff
@@ -209,8 +209,6 @@ func queryRawData(query string) ([]byte, error) {
 		return nil, fmt.Errorf("failed to encode query string: %v", err)
 	}
 
-	defer syscall.RegCloseKey(syscall.HKEY_PERFORMANCE_DATA)
```

Comment:
Why were the calls to close `HKEY_PERFORMANCE_DATA` removed? Are they not necessary?

Reply:
I reading the [Microsoft docs](https://learn.microsoft.com/en-us/windows/win32/perfctrs/using-the-registry-functions-to-consume-counter-data), `RegCloseKey` does not close the Registry call, but frees the DLLs. Since we constantly fetch from registry, I'm not sure if is an good idea to clear the cache and frees the DLLs on each loop. 

Looking at 

* Datadog agent, they do not [call](https://github.com/DataDog/datadog-agent/blob/599c6147dafafbb64772a238c679e1a56b1e0234/pkg/flare/archive_win.go#L43) `syscall.RegCloseKey`, too.
* [Munin](https://github.com/munin-monitoring/munin-node-win32/blob/791d8cd8d54f0d1177d456396f1719761f164930/src/plugins/PerfCounterMuninNodePlugin.cpp#L40), they do also not closing the Handle.
* [pywin32](https://github.com/mhammond/pywin32/blob/e99541164bb65cc76b08804c7a7bbdc52afa47d4/win32/Lib/win32pdhutil.py#L39) they do also not closing the Handle.


There is also a opened PR to remove this RegCloseKeys - https://github.com/leoluk/perflib_exporter/pull/43

Reply:
Thanks, that's good context on the removal